### PR TITLE
Fixed development environment setup commands

### PR DESCRIPTION
**Pull request by ktsrivastava29**
I edited the command in the readme file; previously, there was a $ symbol before it, and when using direct copy, the $ symbol used to get copied as well; moreover, 
```
npm install -g yo @plone/generator-volto 
```
requires sudo access, so sudo was put before it.
Please look into it; if any other changes are needed, I'll be happy to make them.

### CODE REVIEW

File: README.md
Comment (stevepiercy) [inline README.md:65]:
```suggestion
```shell
```

Comment (stevepiercy) [reply README.md:65]: For readability by editors, I would also suggest a blank line before and after the triple backticks for each code block.

File: README.md
Comment (stevepiercy) [inline README.md:70]:
```suggestion
```shell
```

File: README.md
Comment (stevepiercy) [reply README.md:65]: BTW, I agree that `$` should be removed from code blocks. Nowadays the tooling works better without the shell prompt. It also differs across various shells.

File: README.md
Comment (stevepiercy) [inline README.md:66]: `sudo` should not be required to install global dependencies. See https://stackoverflow.com/a/20264272/2214933 and https://medium.com/@ExplosionPills/dont-use-sudo-with-npm-5711d2726aa3. If you must use it, then something is not right with your system configuration. Nonetheless, we need to document what that something is. I don't know what that is at this moment.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Thank you for your PR.

I've requested specifying the language and adding a few blank lines.

Also we really do not want to use `sudo`. Please take a moment to research why it is a really bad idea™.

File: README.md
Comment (ktsrivastava29) [reply README.md]: @stevepiercy Thank you for taking the time to review this PR; I've made some modifications now and would appreciate it if you could have a look at it.

**Comment by tiberiuichim** [comment]
The dollar $ sign is sometimes used to show that some commands are supposed to entered in the shell. I think it's an old habit, probably not needed anymore. The first hit for "bash tutorial" is [filled with these sort of examples](https://linuxconfig.org/bash-scripting-tutorial#:~:text=First%20you%20need%20to%20find%20out%20where%20is%20your%20Bash%20interpreter%20located.%20Enter%20the%20following%20into%20your%20command%20line%3A). I guess the expectation is that the user is familiar with the convention. The other convention, which is to use the # sign for the command prompt is stranger (as it typically denotes a comment), so that's probably why the initial author tried to avoid it.

File: README.md
Comment (stevepiercy) [inline README.md:70]: Please insert a blank line above this sentence.

Also here is a suggestion to take advantage of inline literals and clever way to represent admonitions using plain old Markdown syntax. (h/t https://stackoverflow.com/a/64254203/2214933)

```suggestion
> 💡 **HINT** In case the `npm install` command throws an error, which would be most likely due to a permissions issue in your home directory, you will need to reclaim ownership of the `.npm` directory.
```

Unfortunately the `README.md` does not support MyST syntax because it is a stand alone Markdown document, living separately from the documentation. If it did support MyST syntax, I would use a [`hint` admonition](https://sphinx-themes.org/sample-sites/sphinx-book-theme/kitchen-sink/admonitions/#hint).

File: README.md
Comment (stevepiercy) [inline README.md:71]:
```suggestion
You can do so by executing the following command.
```

File: README.md
Comment (stevepiercy) [inline README.md:77]:
```suggestion
Follow the prompt's questions, providing `myvoltoproject` as the project name.
When it finishes, change your working directory:
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Good stuff! Thank you for the improvements. Let's get one more review from a maintainer. @sneridagh @tiberiuichim

**Comment by stevepiercy** [comment]
@ktsrivastava29 have you submitted your signed  Plone Contributor Agreement? http://docs.plone.org/develop/coredev/docs/contributors_agreement_explained.html

**Comment by stevepiercy** [comment]
> The dollar $ sign is sometimes used to show that some commands are supposed to entered in the shell. I think it's an old habit, probably not needed anymore.

Yup. In Plone Trainings and Documentation, we agreed that the shell prompts are not helpful. They get in the way of copy-paste. It definitely is a habit carried over from plain text documentation that lacks Sphinx, Pygments, and copy-button.

File: README.md
Comment (tiberiuichim) [reply README.md:75]: All this pain can be avoided by the use of NVM, which installs a "local" personalized copy of the nodejs environment. This is documented in the [Volto Docs](https://docs.voltocms.com/getting-started/install/#install-nvm-nodejs-version-manager). Maybe this readme should just point to the docs.

Comment (stevepiercy) [reply README.md:75]: @tiberiuichim great minds think alike. https://github.com/plone/volto/pull/3131#issuecomment-1063315610

Comment (ktsrivastava29) [reply README.md:75]: This sounds to be a good suggestion, should I go on deleting all of these lines and just adding the link to the docs to it?

Comment (stevepiercy) [reply README.md]: I defer to the maintainers to make that call.

**Review by tiberiuichim — APPROVED** [review]
Once final docs for Plone 6 are published, we'll need to update the URLs

**Comment by ktsrivastava29** [comment]
I've changed the docs folder structure that fixes the issue #3093 
Any suggestions for further improvement will appreciated

**Comment by ktsrivastava29** [comment]
I think I should create a new PR for this issue

**Comment by stevepiercy** [comment]
> Once final docs for Plone 6 are published, we'll need to update the URLs

I added two todos just now to remind us to do that thing. Thank you!

https://github.com/plone/documentation/issues/1140

> - [ ] In `conf.py`s in submodules, update the Intersphinx `plone` target to `docs.plone.org` or perhaps with `6` in the host name or URI path.
> - [ ] In `plone/volto`'s `README.md`, update the URLS containg `6.dev-docs.plone.org` to `docs.plone.org`, or perhaps with `6` in the host name or URI path.

**Comment by stevepiercy** [comment]
@ktsrivastava29 it is OK to keep this PR open and moving forward because we have the history of our discussion and why we decided to do the things we did in one place. A new PR would add one more place to look and makes lazy developers (like me) sad. If you're concerned about a long commit history, we can do a [squash merge commit](https://gitbetter.substack.com/p/how-to-squash-git-commits?s=r).

Meanwhile we cannot merge any PR from you unless you have signed the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement). The overly-helpful bot @mister-roboto reminds us with each of your commits until we have received and processed it.

As a side note, which link to the Plone Contributor Agreement would be more helpful for new contributors?

- https://plone.org/foundation/contributors-agreement
- http://docs.plone.org/develop/coredev/docs/contributors_agreement_explained.html (from @mister-roboto)

In either case, we need to update the link that @mister-roboto emits because it uses `http`.

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:188]: Now that we are in the docs instead of the `README.md`, we can use MyST syntax instead of Markdown hacks, and replace the above with the following. Hopefully I got the nested MyST syntax and indentation for an enumerated list item correct. The Netlify preview should help.

`````
    ````{hint}
    In case the `npm install` command throws an error, which would be most likely due to a permissions issue in your home directory, you will need to reclaim ownership of the `.npm` directory.

    You can do so by executing the following command.

    ```shell
    sudo chown -R $(whoami)  ~/.npm
    ```
    ````
`````

File: README.md
Comment (stevepiercy) [inline README.md:66]:
```suggestion
For detailed installation instructions and troubleshooting tips, see the [documentation](https://docs.voltocms.com/getting-started/install/#install-nvm-nodejs-version-manager).
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
We're on the right track now. Thank you for your patience and working through this with us.

**Comment by ktsrivastava29** [comment]
> Meanwhile we cannot merge any PR from you unless you have signed the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement). The overly-helpful bot @mister-roboto reminds us with each of your commits until we have received and processed it.

@stevepiercy I've already emailed the signed pdf, but I'm not sure why the bot still says I haven't signed; I've attached a screenshot for you to look at. 

Screenshot from 2022-03-14 12-54-10

Actually, the documents folder issue #3093 was a little different from this one, so I created a second PR for it #3144 . Should I close that PR and make the changes solely here?

Working with the organisation, I'm learned a lot along the way, I'm grateful that my PR and comments are getting noticed. I look forward to continue working with this.

**Comment by stevepiercy** [comment]
> @stevepiercy I've already emailed the signed pdf, but I'm not sure why the bot still says I haven't signed; I've attached a screenshot for you to look at.

OK, thank you! A human must review your agreement, then press a button somewhere. If it happened on a weekend, please wait until later today. If you do not hear back, then we can escalate it to whoever has that power. Maybe @tkimnguyen knows?

> Actually, the documents folder issue #3093 was a little different from this one, so I created a second PR for it #3144 . Should I close that PR and make the changes solely here?

No. It is an unrelated issue.

**Comment by ktsrivastava29** [comment]
@stevepiercy Thank You, my application has been accepted, can you tell how can this PR be merged

**Comment by ktsrivastava29** [comment]
@stevepiercy now I've added the entry to changelog. I believe this PR is good to get merged

**Comment by tkimnguyen** [comment]
Steve is right: the agreement has to be processed manually and then you would be added to the plone GitHub organization

**Comment by ktsrivastava29** [comment]
@tkimnguyen the agreement has already been processed and now we don't see that bot asking to sign the agreement. 
So I believe we can go on merging this PR to master

File: docs/source/getting-started/install.md
Comment (stevepiercy) [reply docs/source/getting-started/install.md:188]: Please incorporate this suggestion.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by stevepiercy** [comment]
I confirm that the addition of the change log entry removed the blocker of the signed Plone Contributor Agreement. One if the suggestions I made was resolved without action. Please incorporate it, then I can approve. Thank you!

**Comment by ktsrivastava29** [comment]
@stevepiercy I apologize; I made those modifications but somehow didn't save the file, as a result of which it was neither committed nor pushed; now that I've made the necessary adjustments, you can see them.

File: CHANGELOG.md
Comment (stevepiercy) [inline CHANGELOG.md:97]: Volto 15.0.0 was released yesterday. Under the new 15.0.1 heading, would you please:

- create a subheading `### Documentation`, and
- move this item under that subheading?

**Review by tiberiuichim — APPROVED** [review]

File: docs/source/getting-started/install.md
Comment (stevepiercy) [inline docs/source/getting-started/install.md:181]: The indentation here of the backticks appears to be a little bit off. It should align with the "O" in "Open a terminal and execute:" above, its content, and the closing 4 backticks below.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Volto developers are _fast_! 🐎  Just a couple of minor changes and it should be good to go. Thank you!

**Comment by ktsrivastava29** [comment]
@stevepiercy Thank you for pointing out this, I've fixed this now, please have a look

**Review by stevepiercy — APPROVED** [review]
LGTM! Please do the honors of merging your PR.

**Review by sneridagh — COMMENTED** [review]
I'm still not sure of moving these things off the README... Please do not merge yet. We will discuss it in today's Volto's Team meeting.

**Comment by sneridagh** [comment]
Really?

**Comment by sneridagh** [comment]
Sorry, this is mostly my fault for not weighing in, taking into account that I nurture this README since the very beginning. This is the current README *Getting Started* section:

https://github.com/plone/volto#quick-start

For me, a getting started section helps me to really get started in matter of minutes. No further clicking, just test the thing locally in my own machine. If you go now through it, I can't do that. Also:

For detailed installation instructions and troubleshooting tips, see the [documentation](https://docs.voltocms.com/getting-started/install/#install-nvm-nodejs-version-manager).

Points to the install of the version manager? Also the link points to the old docs domain.

Given the importance of today (the day after we released Volto 15), I have to ask you to revert this.

**Comment by sneridagh** [comment]
@ktsrivastava29 you are welcome to open again the PR. However, I think the README is a very special entry point and has to be carefully thought and reviewed. Communication is paramount either for community members and external, potential Plone adopters.

If you take a look at other project's Quick Start guide tend to be concise and directly to the point. Volto is a bit more complex to setup, so the section should be long for sure. 

We should move other parts to the docs? I think we could, training, talks, videos, development directions, etc. 

Let's continue the discussion about it.

**Comment by ktsrivastava29** [comment]
@sneridagh Thank you, soon I'll be opening a PR again and we can have a discussion over there.

**Pull request merged**
